src/qibolab/instruments/qm/config.py

```diff
@@ -189,7 +189,26 @@ def register_flux_element(self, qubit, intermediate_frequency=0):
         else:
             self.elements[f"flux{qubit.name}"]["intermediate_frequency"] = intermediate_frequency
 
-    def register_pulse(self, qubit, pulse, time_of_flight, smearing):
+    def register_element(self, qubit, pulse, time_of_flight=0, smearing=0):
+        if pulse.type is PulseType.DRIVE:
+            # register drive element
+            if_frequency = pulse.frequency - math.floor(qubit.drive.local_oscillator.frequency)
+            self.register_drive_element(qubit, if_frequency)
+            # register flux element (if available)
+            if qubit.flux:
+                self.register_flux_element(qubit)
+        elif pulse.type is PulseType.READOUT:
+            # register readout element (if it does not already exist)
+            if_frequency = pulse.frequency - math.floor(qubit.readout.local_oscillator.frequency)
+            self.register_readout_element(qubit, if_frequency, time_of_flight, smearing)
+            # register flux element (if available)
+            if qubit.flux:
+                self.register_flux_element(qubit)
+        else:
+            # register flux element
+            self.register_flux_element(qubit, pulse.frequency)
```

Comment:
I think here you are supposing that this is a PulseType.FLUX, but would this work also for PulseType.FLUX_COUPLER?